### PR TITLE
OneBranch Migration

### PR DESCRIPTION
**Pull request by trwalke**
# OneBranch Migration
Enabling onebranch builds

- [ ] You've read the [Contributor Guide](https://github.com/AzureAD/azure-activedirectory-identitymodel-extensions-for-dotnet/blob/dev/Contributing.md) and [Code of Conduct](https://opensource.microsoft.com/codeofconduct/).
- [ ] You've included unit or integration tests for your change, where applicable.
- [ ] You've included inline docs for your change, where applicable.
- [ ] If any gains or losses in performance are possible, you've included benchmarks for your changes. [More info](https://github.com/AzureAD/azure-activedirectory-identitymodel-extensions-for-dotnet/wiki/Benchmarking-in-Wilson)
- [ ] There's an open issue for the PR that you are making. If you'd like to propose a new feature or change, please open an issue to discuss the change or find an existing issue.

Summary of the changes (Less than 80 chars)

## Description

Enabling OneBranch Build
The WilsonSourceDirectory variable represents the location of the src folder on OneBranch pipelines. This is set in the pipeline UI.

PipelineType is also set on the pipeline UI to differentiate between the legacy and onebranch pipelines

SDL tasks are handled by OneBranch so they are disabled in the OneBranch pipeline

Fixes #{bug number} (in this specific format)

Working Build: https://dev.azure.com/IdentityDivision/IDDP/_build/results?buildId=1290552&view=results

### CODE REVIEW

File: build/template-Build-run-tests-sign.yml [hunk 1]
```diff
@@ -0,0 +1,213 @@
...
+              "operationSetCode": "NuGetVerify",
+              "parameters": [ ],
+              "toolName": "sign",
+              "toolVersion": "1.0"
+          }
+      ]
+    SessionTimeout: 20
+  condition: and(succeeded(), eq(variables['PipelineType'], 'legacy'))
+
+- task: onebranch.pipeline.signing@1
```
Comment (trwalke) [inline build/template-Build-run-tests-sign.yml:158]: enabling OneBranch signing. Sign validation is automatically injected into the OneBranch build pipeline.

File: build/template-Build-run-tests-sign.yml [hunk 2]
```diff
@@ -0,0 +1,213 @@
...
+- task: securedevelopmentteam.vss-secure-development-tools.build-task-publishsecurityanalysislogs.PublishSecurityAnalysisLogs@3
+  displayName: 'Publish Security Analysis Logs'
+  continueOnError: true
+
+- task: securedevelopmentteam.vss-secure-development-tools.build-task-postanalysis.PostAnalysis@2
+  displayName: 'Post SDL Analysis'
+  continueOnError: true
+
+#Sign Wilson 7x task group
+- template: template-sign-wilson.yaml
```
Comment (bgavrilMS) [inline build/template-Build-run-tests-sign.yml:115]: Why did you split them in templates? I do not see these 2 templates being used more than once...

File: build/template-Build-run-tests-sign.yml [hunk 3]
```diff
@@ -0,0 +1,213 @@
...
+    packDirectory: '$(Build.SourcesDirectory)\artifacts'
+    nobuild: true
+    verbosityPack: 'Minimal'
+    configuration: $(BuildConfiguration)
+    packagesToPack: '$(WilsonSourceDirectory)Product.proj'
+    feedsToUse: 'config'
+    nugetConfigPath: Nuget.config
+    externalFeedCredentials: 'Internal Analyzers'
+
+- task: EsrpCodeSigning@4
```
Comment (bgavrilMS) [inline build/template-Build-run-tests-sign.yml:131]: Is this still needed if ESRP code signing is still on?

File: build/template-Build-run-tests-sign.yml (hunk 2 above)
Comment (trwalke) [reply build/template-Build-run-tests-sign.yml:115]: I didn't split them. this is how it was from the beginning. I can do some additional refactoring to make things simpler.

File: build/template-Build-run-tests-sign.yml (hunk 3 above)
Comment (trwalke) [reply build/template-Build-run-tests-sign.yml:131]: Not necessarily, I am just keeping the old build functional until we finally pull the plug on it.

Comment (bgavrilMS) [reply build/template-Build-run-tests-sign.yml:131]: Got it, I see the "legacy" flag now, thanks. Maybe add a comment.

**Review by bgavrilMS — APPROVED** [review]

**Review by FuPingFranco — APPROVED** [review]
LGTM! Thanks @trwalke

**Comment by FuPingFranco** [comment]
@trwalke Could you update the official release guidelines https://identitydivision.visualstudio.com/DevEx/_git/MISE?path=/docs/releases/OfficialReleases.md&version=GBmaster&_a=previewonce once both PR's are in and tested in master. Thanks!

**Pull request merged**